hubspot-style/src/test/java/com/hubspot/immutables/utils/WireSafeEnumTest.java

```diff
@@ -300,6 +308,42 @@ public void itBuildsFromKnownStringWithCollidingJsonAndCreator() {
     assertThat(wrapper.asEnum()).isEqualTo(Optional.of(CollidingJsonEnumWithCreator.DEF));
   }
 
+  @Test
+  public void itBuildsFromJsonAlias() {
```

Comment:
could you add an assertion for the JSON representation of the wrapper? I'd expect it to be `"ABC"` instead of the asString but I'd like to see that tested

Reply:
The serialized JSON value will actually be `"abc"` (the alias) here, since `asString` is used as the `@JsonValue`. I can use the non-alias name in the jsonValue when writing into the cache, but doing that feels a little odd because it won't match the JSON string that we deserialized from anymore.

Reply:
hmm I'm not sure how I feel about this lol

Reply:
how does a bare enum work with `@JsonAlias`? If it's serialized using the actual constant name, I think we should prefer that behavior

Reply:
Yep, it'll use the constant name - I'll update the behavior to reflect that.

Tested with the following
```java
  enum TestEnum {
    @JsonAlias("abc")
    ABC,
  }

  @Test
  void testJsonAliasRoundtrip() throws JsonProcessingException {
    ObjectMapper objectMapper = new ObjectMapper();
    TestEnum val = objectMapper.readValue("\"abc\"", TestEnum.class);
    System.out.println(objectMapper.writeValueAsString(val));
  }
```
Output:
```
"ABC"
```

Reply:
The "most correct / consistent with jackson" behavior seems (to me) to be that if we deserialize and resolve a real enum constant, then the json value emitted during subsequent re-serialization should be the non-aliased (primary) json representation.  But that also is somewhat counter to the way wiresafe enum works today, and it runs counter to how existing implementations behave with `WireSafeEnum` when utilizing a `@JsonCreator` that maps multiple json values to a single enum instance.  For example

```java
public enum TestEnum {
  ONE("one");

  private static final Map<String, TestEnum> LOOKUP = Maps.uniqueIndex(
    Arrays.asList(values()),
    TestEnum::getValue
  );
  private final String value;

  TestEnum(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @JsonCreator
  public static TestEnum fromValue(String value) {
    return LOOKUP.get(value.toLowerCase());
  }
}

public static class TestObject {

  @JsonProperty("value")
  final WireSafeEnum<TestEnum> value;

  public TestObject(@JsonProperty("value") WireSafeEnum<TestEnum> value) {
    this.value = value;
  }
}

@Test
public void testJson() {
  TestObject object = JsonUtils.readValue(
    "{\"value\": \"oNe\"}",
    new TypeReference<>() {}
  );
}
```
<img width="569" alt="Screenshot 2024-04-18 at 3 53 34 PM" src="https://github.com/HubSpot/hubspot-immutables/assets/2109558/d5984458-da82-4d6e-9ecd-69ab2fc8d763">

All that is to say while storing the alias value as a json value feels less correct to me, it's also the most consistent with what people are already doing.

Reply:
Thinking about this more, I think i'm coming down more strongly on the side of "preserve the received string as is", since that seems (to me) to be more likely to result in the expected behavior in instances where you're communicating through a chain of services/deployables where some of them may not know about the new aliasing setup yet 🤔